cg/cli/workflow/balsamic/base.py

```diff
@@ -1,323 +1,263 @@
+        LOG.warning(f"No config file found for {case_id}!")
+        raise click.Abort()
+
+    deliverables_file_path = context.obj["BalsamicAnalysisAPI"].get_deliverables_file_path(case_id)
+    if not Path(deliverables_file_path).exists():
+        LOG.warning(f"No deliverables file found for {case_id}")
+        raise click.Abort()
+
+    try:
+        context.obj["BalsamicAnalysisAPI"].update_housekeeper(
+            case_object=case_object,
+            sample_config=sample_config,
+            deliverables_file_path=deliverables_file_path,
+        )
+        context.obj["BalsamicAnalysisAPI"].update_statusdb(
+            case_object=case_object, sample_config=sample_config,
+        )
+    except BundleAlreadyAddedError as e:
+        LOG.warning(f"Could not store bundle in Housekeeper and StatusDB: {e.message}!")
+        raise click.Abort()
+
+
+@balsamic.command("start")
+@ARGUMENT_CASE_ID
+@OPTION_ANALYSIS_TYPE
+@OPTION_PRIORITY
+@OPTION_RUN_ANALYSIS
+@OPTION_DRY
+def start(context, case_id, analysis_type, priority, run_analysis, dry):
+    """Start full workflow for CASE ID"""
+    context.invoke(link, case_id=case_id)
+    context.invoke(config_case, case_id=case_id, dry=dry)
+    sleep(10)
+    context.invoke(
+        run,
+        case_id=case_id,
+        analysis_type=analysis_type,
+        priority=priority,
+        run_analysis=run_analysis,
+        dry=dry,
+    )
 
-        if dry_run:
-            continue
 
+@balsamic.command("start-available")
+@OPTION_DRY
+def start_available(context, dry):
+    """Start full workflow for all available cases"""
+    for case_object in context.obj["BalsamicAnalysisAPI"].store.cases_to_balsamic_analyze():
+        case_id = case_object.internal_id
+        LOG.info(f"Starting analysis for {case_id}")
         try:
-            context.invoke(balsamic, priority=priority, case_id=case_obj.internal_id)
-        except LimsDataError as error:
-            LOG.exception(error.message)
-            exit_code = FAIL
-
-    sys.exit(exit_code)
-
-
-@balsamic.command("remove-fastq")
-@click.option("-c", "--case", "case_id", help="remove fastq folder for a case")
-@click.pass_context
-def remove_fastq(context, case_id):
-    """Remove case fastq folder"""
+            context.invoke(start, case_id=case_id, dry=dry, run_analysis="--run-analysis")
+        except click.Abort():
+            continue
 
-    wrk_dir = Path(f"{context.obj['balsamic']['root']}/{case_id}/fastq")
 
-    if wrk_dir.exists():
-        shutil.rmtree(wrk_dir)
+@balsamic.command("deliver-available")
+@OPTION_DRY
+def deliver_available(context, dry):
+    """Deliver bundle data for all available cases"""
+    for case_object in context.obj["BalsamicAnalysisAPI"].store.analyses_to_deliver():
+        case_id = case_object.internal_id
+        LOG.info(f"Storing analysis for {case_id}")
+        try:
+            context.invoke(report_deliver, case_id=case_id, dry=dry)
+            sleep(10)
```

Comment:
do we need a sleep here? It seems a bit hacky to me, can it be solved in any other way?

Reply:
I agree, this might have better solutions.
This might work just fine without, but I there is a risk it will be a minor problem. I have used sleep to pause the cli before submitting next command, but only for commands which call balsamic through Process. When process calls balsamic command, it immediately can continue constructing and submitting the next one, which would require the previous step to be finished in balsamic. If it is not finished, we would receive an error. This was not a problem with bash solution previously, but with process it is now possible to be too fast in some cases